packages/server/controllers/images.js

```diff
@@ -0,0 +1,40 @@
+const { v4: uuidv4 } = require("uuid");
+const { getSignedUrl } = require("@aws-sdk/s3-request-presigner");
+const { s3Client } = require("../utils/s3");
+const { PutObjectCommand, GetObjectCommand } = require("@aws-sdk/client-s3");
+
+async function upload(imageType, imageData) {
+  const id = uuidv4();
```

Comment:
You should not generate a new UUID here. The UUID should come from the frontend in the request body along with the image type and data. Pass the UUID as a parameter to the upload function here to use for the key.